src/form-elements/header/donation-form-header.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  LitElement,
+  html,
+  css,
+  customElement,
+  CSSResult,
+  TemplateResult,
+  property,
+  query,
+} from 'lit-element';
+
+import './donation-summary';
+import './edit-donation';
+import { DonationPaymentInfo } from '../../models/donation-info/donation-payment-info';
+import { EditDonation } from './edit-donation';
+
+export enum DonationFormHeaderMode {
+  Summary = 'summary',
+  Edit = 'edit',
+}
+
+@customElement('donation-form-header')
+export class DonationFormHeader extends LitElement {
+  @property({ type: Object }) donationInfo: DonationPaymentInfo = DonationPaymentInfo.default;
+
+  @property({ type: String }) mode: DonationFormHeaderMode = DonationFormHeaderMode.Edit;
+
+  @query('edit-donation') editDonation?: EditDonation;
+
+  @query('donation-summary') donationSummary?: EditDonation;
+
+  /** @inheritdoc */
+  render(): TemplateResult {
+    return html`
+      ${this.currentTemplate}
+    `;
+  }
+
+  private get currentTemplate(): TemplateResult {
+    switch (this.mode) {
+      case DonationFormHeaderMode.Summary:
+        return this.donationSummaryTemplate;
+      case DonationFormHeaderMode.Edit:
+        return this.editDonationTemplate;
```

Comment:
Since `this.mode` is an `enum` of type `DonationFormHeaderMode`, TypeScript won't allow any other values except for these to be set. Every interaction with it would be like `this.mode = DonationFormHeaderMode.Summary` instead of assigning a string value to it like `this.mode = 'summary'`. 

The other cool thing about it being an `enum` is that if we ever add another option to it, say `DonationFormHeaderMode.OpenLibrary`, TypeScript will alert you that you haven't accounted for all of the possible options here.